src/MSBuild/XMake.cs

```diff
@@ -3552,8 +3552,8 @@ internal static string ProcessProjectSwitch(
                     }
                 }
 
-                // Get all files in the current directory that have a sln extension
-                string[] potentialSolutionFiles = getFiles(projectDirectory ?? ".", "*.sln");
+                // Get all files in the current directory that have a sln or slnx extension
+                string[] potentialSolutionFiles = getFiles(projectDirectory ?? ".", "*.sln?");
```

Comment:
This would match SLNF solution filter files too. Even if that's the intent (the comment above indicates otherwise), it might also inadvertently match any future file extensions that might start with SLN and end with another letter other than X and F. I think explicit matching would be better even if SLNF are included.

Reply:
It is the intent: this is "get all the potential solution filenames", then in the `foreach` below the code filters the list to only what this code cares about ("is solution" and "is solution filter").

Reply:
@rainersigwald Then I propose the comment to reflect that. :)